robottelo/host_helpers/contenthost_mixins.py

```diff
@@ -90,11 +90,16 @@ def _dogfood_helper(self, product, release, repo=None):
             )
         return product, release, v_major, repo
 
-    def download_repofile(self, product=None, release=None, snap=''):
+    def download_repofile(self, product=None, release=None, snap='', proxy=None):
         """Downloads the tools/client, capsule, or satellite repos on the machine"""
         product, release, v_major, _ = self._dogfood_helper(product, release)
-        url = dogfood_repofile_url(settings.ohsnap, product, release, v_major, snap)
-        self.execute(f'curl -o /etc/yum.repos.d/dogfood.repo -L {url}')
+        if not proxy and settings.server.is_ipv6:
+            proxy = settings.server.http_proxy_ipv6_url
+        url = dogfood_repofile_url(settings.ohsnap, product, release, v_major, snap, proxy=proxy)
+        command = f'curl -o /etc/yum.repos.d/dogfood.repo -L {url}'
+        if settings.server.is_ipv6:
+            command += f' -x {settings.server.http_proxy_ipv6_url}'
+        self.execute(command)
```

Comment:
Its already there used in `_dogfood_helper` function to not surprise :)

The only way we have is to override the function in Satellite class, because though its defined in ContentHost class its used for Satellite in this case.

Reply:
Also there are many places in Host classes which are not adhering to the class in which context they are. The best way to clean is to override the method in child class and call super method.

For this PR I don't want to sign up for that work, we could see it as a separate effort.